src/databricks/labs/ucx/hive_metastore/catalog_schema.py

```diff
@@ -44,162 +40,81 @@ def create_ucx_catalog(self, prompts: Prompts, *, properties: dict[str, str] | N
             properties : (dict[str, str] | None), default None
                 The properties to pass to the catalog. If None, no properties are passed.
         """
-        self._create_catalog_validate(self._ucx_catalog, prompts, properties=properties)
+        self._create_catalog_validate(Catalog(self._ucx_catalog), prompts, properties=properties)
 
     def create_all_catalogs_schemas(self, prompts: Prompts) -> None:
-        candidate_catalogs, candidate_schemas = self._get_missing_catalogs_schemas()
-        for candidate_catalog in candidate_catalogs:
-            self._create_catalog_validate(candidate_catalog, prompts, properties=None)
-        for candidate_catalog, schemas in candidate_schemas.items():
-            for candidate_schema in schemas:
-                try:
-                    self._create_schema(candidate_catalog, candidate_schema)
-                except BadRequest as e:
-                    if "already exists" in str(e):
-                        logger.warning(f"Skipping already existing schema: {candidate_catalog}.{candidate_schema}")
-                        continue
-        self._apply_from_legacy_table_acls()
-        self._update_principal_acl()
-
-    def _apply_from_legacy_table_acls(self) -> None:
-        grants = self._get_catalog_schema_hive_grants()
-        for grant in grants:
-            acl_migrate_sql = grant.uc_grant_sql()
-            if acl_migrate_sql is None:
-                logger.warning(
-                    f"Skipping legacy grant that is not supported in UC: {grant.action_type} on {grant.this_type_and_key()}"
-                )
+        catalogs_existing, schemas_existing = self._catalogs_schemas_from_unity_catalog()
+        catalogs, schemas = self._catalogs_schemas_from_table_mapping()
+        for src_schema, dst_catalog in catalogs:
+            if dst_catalog in catalogs_existing:
```

Comment:
@nfx and @FastLee : This introduces the question on "do we want to skip applying ACLs when the catalog exists?" 

Before this PR:
- ACLs might be applied to an existing catalog if a schema will be created within that catalog
- ACLs will be skipped for an existing schema.

Reply:
I would skip.
Let's just alert.

Reply:
Skip what exactly? The ones for the catalog if the catalog exists, the ones for the schema if the schema exits, the ones for the catalog if the schema exits? All ACLs if any schema or catalog exists?

Reply:
Currently implemented that **all** legacy grants are migrated to UC and created [an issue](https://github.com/databrickslabs/ucx/issues/2964) to make this part optional.